src/App.tsx

```diff
@@ -15,71 +15,86 @@ export const goods = [
-      />
-    </h1>
+  selectBtn = (item: string) => {
+    const { selectedGood } = this.state;
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+    if (selectedGood !== item) {
+      this.setState({ selectedGood: item });
+    } else {
+      this.setState({ selectedGood: '' });
+    }
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  render() {
+    const { selectedGood } = this.state;
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+    return (
+      <main className="section container">
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+        {selectedGood
+          ? (
+            <h1 className="title is-flex is-align-items-center">
+              {`${selectedGood} is selected`}
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={this.clearGood}
+              />
+            </h1>
+          )
+          : <h1 className="title">No goods selected</h1>}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+        <table className="table">
+          <tbody>
+            {goods.map(good => (
+              <tr
+                data-cy="Good"
+                key={good}
+                className={
+                  selectedGood === good
+                    ? 'has-background-success-light'
+                    : undefined
+                }
+              >
+                <td>
+                  <button
+                    data-cy="AddButton"
+                    type="button"
+                    className={
+                      selectedGood === good
+                        ? 'button is-info'
+                        : 'button'
+                    }
+                    onClick={() => this.selectBtn(good)}
+                  >
+                    {
+                      selectedGood === good
+                        ? '-'
+                        : '+'
+                    }
```

Comment:
looks like a crutch a little. I would create two separate buttons. Readability of the code is our first priority

Reply:
You meant, I need:
{ isSelected
  ? <button
        data-cy="AddButton"
        type="button"
        className='is-info'
        onClick={() => this.selectButton(good)}
      >
        -
      </button>
  : <button
        data-cy="AddButton"
        type="button"
        onClick={() => this.selectButton(good)}
      >
        +
      </button>
}

Reply:
yes